### PR TITLE
[3.x] clean cache after enabling 2fa in postinstall

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #30398  .

### Summary of Changes
redo of #30416 on staging
clean cache after enabling 2fa in postinstall
see https://github.com/joomla/joomla-cms/pull/30416#issuecomment-675958570

### Testing Instructions
see #30398 but on 3.x

### Actual result BEFORE applying this Pull Request
2fa not enabled

### Expected result AFTER applying this Pull Request
2fa enabled

### CODE REVIEW

File: plugins/twofactorauth/totp/postinstall/actions.php [hunk 1]
```diff
@@ -51,12 +51,14 @@ function twofactorauth_postinstall_action()
...
-		->set($db->qn('enabled') . ' = ' . $db->q(1))
+		->set($db->qn('enabled') . ' = 1')
 		->where($db->qn('type') . ' = ' . $db->q('plugin'))
 		->where($db->qn('folder') . ' = ' . $db->q('twofactorauth'));
 	$db->setQuery($query);
 	$db->execute();
 
+	// Clean cache.
+	JFactory::getCache()->clean('com_plugins');
 	// Redirect the user to their profile editor page
```
Comment (richard67) [inline plugins/twofactorauth/totp/postinstall/actions.php:63]:
```suggestion

	// Redirect the user to their profile editor page
```
PHPCS

**Comment by Razzo1987** [comment]
Sorry I test it on J4.... and work correctly
immagine
immagine

**Comment by Razzo1987** [comment]
I have not tested this item.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30419">issues.joomla.org/tracker/joomla-cms/30419</a>.</sub>

**Comment by adj9** [comment]
I have tested this item :white_check_mark: successfully on c1c810107f0d6b360e6dd817927dabcc32d94723<br /><br />Done.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30419">issues.joomla.org/tracker/joomla-cms/30419</a>.</sub>

**Comment by Razzo1987** [comment]
with Joomla! 3.9.21 the button in the post install messages enable 2fa without the patch<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30419">issues.joomla.org/tracker/joomla-cms/30419</a>.</sub>

**Comment by adj9** [comment]
I have not tested this item.<br /><br />On Joomla 3.9 and 3.10 it doesn't work even with the cache enabled, on Joomla 4 it has no problems.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30419">issues.joomla.org/tracker/joomla-cms/30419</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on c1c810107f0d6b360e6dd817927dabcc32d94723<br /><br />System Cache: ON - Conservative caching<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30419">issues.joomla.org/tracker/joomla-cms/30419</a>.</sub>

**Comment by BertaOctech** [comment]
I have tested this item :white_check_mark: successfully on c1c810107f0d6b360e6dd817927dabcc32d94723<br /><br />I take a J!4 fresh install (beta 5)
I activate progressive cahing

I activate 2F autentication from link in postinstall messages
As stated, it redirects to the user configuration and 2Fa options tab is not present 

Now I deactivate 2 factor autentications plugins which were activated from message, so the button is available again

I apply patch, even if it is labeled as [3.x]

Now, after when I click on Activate 2 factor autentication, I am redirected to the user configuration and the tab appears<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30419">issues.joomla.org/tracker/joomla-cms/30419</a>.</sub>

**Comment by pabloarias** [comment]
I have tested this item :white_check_mark: successfully on c1c810107f0d6b360e6dd817927dabcc32d94723<br /><br />I've tested on Joomla 4-beta2-dev and PHP 7.3.22-1.

Note: after applying path, clicking in message button to enable 2FA, it redirects to user profile and 2FA is enabled, perfect. But it does not redirects to 2FA tab, as it happens when cache is not enabled.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30419">issues.joomla.org/tracker/joomla-cms/30419</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30419">issues.joomla.org/tracker/joomla-cms/30419</a>.</sub>

**Comment by alikon** [comment]
thanks all for the tests, i'm happy that this work on j4 too....
but this pr has been requested on the j3 branch.... so if you have some more spare time please test on staging too

**Comment by zero-24** [comment]
Merging thanks 👍

**Pull request merged**